### PR TITLE
refactoring to prepare for RPC API

### PR DESCRIPTION
**Pull request by jleni**
This PR is related to https://github.com/goerli/parity-goerli/pull/58 and https://github.com/goerli/parity-goerli/issues/35

As discussed, to simplify reviews, we will early merge some of the improvements and refactoring changes. This is in preparation to expose the engine int the RPC API.

### CODE REVIEW

File: ethcore/src/engines/clique/mod.rs [hunk 1]
```diff
@@ -256,93 +255,104 @@ impl Clique {
...
+		Ok(new_state)
+	}
+
+	pub fn find_header(&self, query: BlockId) -> Result<Header, Error> {
+		let client = self.client.read().as_ref().and_then(|w| w.upgrade()).ok_or(EngineError::RequiresClient)?;
+		let header = client.block_header(query).ok_or_else(|| "block not found")?.decode()?;
+		Ok(header)
+	}
+
+	pub fn get_signers(&self, query: BlockId) -> Result<Vec<Address>, Error> {
```
Comment (jwasinger) [inline ethcore/src/engines/clique/mod.rs]: I think this method should wrap `state` (https://github.com/goerli/parity-goerli/blob/master/ethcore/src/engines/clique/mod.rs#L258) to get the signers for a given block hash.  The full signers list is only present in the header at the epoch transition.

File: ethcore/src/engines/clique/mod.rs [hunk 2]
```diff
@@ -501,21 +511,21 @@ impl Engine<EthereumMachine> for Clique {
 					// Wait for the right moment.
 					if now < limit {
 						trace!(target: "engine",
-								"generate_seal: sleeping to sign: inturn: {}, now: {:?}, to: {:?}.",
-								inturn, now, limit);
+							   "generate_seal: sleeping to sign: inturn: {}, now: {:?}, to: {:?}.",
```
Comment (jwasinger) [inline ethcore/src/engines/clique/mod.rs]: Should revert these changes.  They introduce spaces.

File: ethcore/src/engines/clique/mod.rs (hunk 1 above)
Comment (jleni) [reply ethcore/src/engines/clique/mod.rs]: done

**Comment by jleni** [comment]
It would be good to review/merge this initial part so it is easier to move forward.

**Comment by soc1c** [comment]
we are about to merge clique upstream, feel free to base your work on this but for now I have to put it on ice until the first PR is merged, sorry

**Comment by jleni** [comment]
> we are about to merge clique upstream, feel free to base your work on this but for now I have to put it on ice until the first PR is merged, sorry

No problem. I will close this and continue in my fork as there are probably many other changes that I need to make anyway.
